### ISSUE TITLE
Missing Markdown

### ISSUE THREAD

**Issue by poudink** [uowuo/abaddon#58]
Code markdown seems to be unimplemented, with and without syntax highlighting.
Abaddon:
![image](https://user-images.githubusercontent.com/29778535/156861709-2ef5464f-4cd8-4c16-98f0-0cabe5b72857.png)
Discord:
![image](https://user-images.githubusercontent.com/29778535/156861806-f78521e4-c4ef-4955-ab1e-98aa362f76e5.png)

Hypertext too (here combined with code markdown). That's an embed only feature tho:
Abaddon:
![image](https://user-images.githubusercontent.com/29778535/156861759-a40b0a0b-c013-4f57-ada9-4924225d2a65.png)

Discord:
![image](https://user-images.githubusercontent.com/29778535/156861885-cd4d3ffe-37ee-43c5-a82c-8a8d57ede1f9.png)

EDIT: Hadn't even noticed, but it seems like pretty much no markdown is supported. Thought those two were outliers.

**Comment by ouwou** [comment]
syntax highlighting probably wont happen (for a while) since the only viable option (i know of) for it would be to use gtksourceview which would only be an option on mingw64 builds which i am working on but i dont know when ill actually make the switch.

ive been putting off markdown too since text parsing seems scary but i havent really tried to do it, ill have a look soon™

**Comment by poudink** [comment]
looks like you switched to mingw a while ago, now

**Comment by ouwou** [comment]
yeah i tried to do it a couple weeks ago but it didnt play nice with other text parsing going on. ill revisit it once i get through some other things that ive wanted to get done for a bit too long

**Comment by 9xN** [comment]
> yeah i tried to do it a couple weeks ago but it didnt play nice with other text parsing going on. ill revisit it once i get through some other things that ive wanted to get done for a bit too long

this will be great looking forward to it!

**Comment by KaKi87** [comment]
Hello,
Any news on this ?
Thanks

**Comment by techflashYT** [comment]
I would also love to see some text processing capabilities, at least the basic `inline code`, *italics*, **bold**, ***bold italics***, and
```
triple-backtick codeblocks
with proper multiline support
```

**Comment by Derpford** [comment]
Aren't there, like, a ton of Markdown libraries? It should be pretty easy to find something that handles parsing the text into an abstract syntax tree for you...

**Comment by ouwou** [comment]
> Aren't there, like, a ton of Markdown libraries? It should be pretty easy to find something that handles parsing the text into an abstract syntax tree for you...

abaddons is really scuffed. in [acheron](https://github.com/ouwou/acheron) i still have to run my own but its better. the goal is to match discord markdown parsing 1:1 and the only real way to accomplish that is reimplementing it from scratch

**Comment by KaKi87** [comment]
> the goal is to match discord markdown parsing 1:1 and the only real way to accomplish that is reimplementing it from scratch

Actually, with the help of [Etorix](https://github.com/EtorixDev) from [Equicord](https://github.com/equicord/equicord), I was able to use the modded client's custom API to find the actual code that Discord uses to parse Discord-flavored Markdown to HTML :

```js
const { markdownToHtml } = Vencord.Webpack.find(m => !!m.markdownToHtml);
markdownToHtml('Hello, **World**!'); // -> '<div class="paragraph">Hello, <strong>World</strong>!</div>'
```

<img width="511" height="86" alt="Image" src="https://github.com/user-attachments/assets/52ae9a8e-e35c-4993-93ef-2782516865b1" />

I published the source code here : https://git.kaki87.net/KaKi87/discord-markdown-to-html

As my use case was just coding an anti-spam bot that works against the [crazy formatting](https://github.com/user-attachments/assets/4c769cae-f544-434d-967e-738fe251f0fa) that they allow, I didn't bother to create any documentation after my integration succeeded, but I can imagine someone who needs to reproduce this parser for a JS-free alternative frontend for Discord could port this minified code into C++ for example by having an LLM generate unit tests.

**Comment by ouwou** [comment]
i mean technically its https://github.com/Khan/perseus/tree/main/packages/simple-markdown with some extra rules but it being javascript and these projects being c++ is basically what i meant by having to reimplement from scratch. i started here https://github.com/ouwou/acheron/blob/master/src/Core/Markdown/Parser.cpp

**Comment by KaKi87** [comment]
How do you know they're using that one ?

**Comment by ouwou** [comment]
you can compare the minified code with the repo and its clearly it. they also credit it explicitly in their open source page

**Comment by KaKi87** [comment]
Oh...
Well, in case anyone is considering it, don't use that library for any project, cause its parsing mechanism is exploitable in dozens of ways.
I'm gonna be submitting tons of reports.
